## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 524182`
Commit: [`f0ed80602b11a0f0808c3d6f26e5ca4895bf5ea9`](https://github.com/NixOS/nixpkgs/commit/f0ed80602b11a0f0808c3d6f26e5ca4895bf5ea9) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/f0ed80602b11a0f0808c3d6f26e5ca4895bf5ea9..pull/524182/head))
Merge: [`400dbe6c5ed6ad4a910c05ffc020d5caa0074a9b`](https://github.com/NixOS/nixpkgs/commit/400dbe6c5ed6ad4a910c05ffc020d5caa0074a9b)

Logs: https://github.com/SkohTV/nixpkgs-review-gha/actions/runs/26418733880


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>pew</li>
    <li>pipenv</li>
    <li>virtualenv-clone (python313Packages.virtualenv-clone)</li>
    <li>python313Packages.virtualenvwrapper</li>
    <li>python314Packages.virtualenv-clone</li>
    <li>python314Packages.virtualenvwrapper</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>pew</li>
    <li>pipenv</li>
    <li>virtualenv-clone (python313Packages.virtualenv-clone)</li>
    <li>python313Packages.virtualenvwrapper</li>
    <li>python314Packages.virtualenv-clone</li>
    <li>python314Packages.virtualenvwrapper</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pipenv</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>pew</li>
    <li>virtualenv-clone (python313Packages.virtualenv-clone)</li>
    <li>python313Packages.virtualenvwrapper</li>
    <li>python314Packages.virtualenv-clone</li>
    <li>python314Packages.virtualenvwrapper</li>
  </ul>
</details>

---
<details>
<summary>pipenv</summary>
<pre>E       PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/var&#x27;

/nix/store/fyyi8b1dpk3xpjnn52rznnpvlyivds97-python3-3.13.13/lib/python3.13/pathlib/_local.py:515: PermissionError
=============================== warnings summary ===============================
../../../store/2xqabipipmxak0dj6sqx5bn8kbf46xzg-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
../../../store/2xqabipipmxak0dj6sqx5bn8kbf46xzg-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
../../../store/2xqabipipmxak0dj6sqx5bn8kbf46xzg-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
../../../store/2xqabipipmxak0dj6sqx5bn8kbf46xzg-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/2xqabipipmxak0dj6sqx5bn8kbf46xzg-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: plugins
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
SKIPPED [6] tests/unit/test_utils.py:444: Windows file paths tested
SKIPPED [1] tests/unit/test_utils.py:651: Windows test only
SKIPPED [1] tests/unit/test_utils_windows_executable.py:15: Windows test only
SKIPPED [1] tests/unit/test_utils_windows_executable.py:30: Windows test only
FAILED tests/unit/test_pylock.py::test_load_lockfile_without_pipfile - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/var&#x27;
============= 1 failed, 409 passed, 9 skipped, 4 warnings in 4.06s =============</pre>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pipenv</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>pew</li>
    <li>virtualenv-clone (python313Packages.virtualenv-clone)</li>
    <li>python313Packages.virtualenvwrapper</li>
    <li>python314Packages.virtualenv-clone</li>
    <li>python314Packages.virtualenvwrapper</li>
  </ul>
</details>

---
<details>
<summary>pipenv</summary>
<pre>E       PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/var&#x27;

/nix/store/ygxqin6ydzjfawywqpp5pal8wv6sf5bh-python3-3.13.13/lib/python3.13/pathlib/_local.py:515: PermissionError
=============================== warnings summary ===============================
../../../store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
../../../store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
../../../store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
../../../store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434
  /nix/store/bx9hk4sq4gl0zknirz31kfknkzdfnlf2-python3.13-pytest-9.0.3/lib/python3.13/site-packages/_pytest/config/__init__.py:1434: PytestConfigWarning: Unknown config option: plugins
  
    self._warn_or_fail_if_strict(f&quot;Unknown config option: {key}\n&quot;)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
SKIPPED [6] tests/unit/test_utils.py:444: Windows file paths tested
SKIPPED [1] tests/unit/test_utils.py:651: Windows test only
SKIPPED [1] tests/unit/test_utils_windows_executable.py:15: Windows test only
SKIPPED [1] tests/unit/test_utils_windows_executable.py:30: Windows test only
FAILED tests/unit/test_pylock.py::test_load_lockfile_without_pipfile - PermissionError: [Errno 1] Operation not permitted: &#x27;/nix/var&#x27;
============= 1 failed, 409 passed, 9 skipped, 4 warnings in 1.50s =============</pre>
</details>

--- a/pkgs/development/python-modules/virtualenv-clone/default.nix
+++ b/pkgs/development/python-modules/virtualenv-clone/default.nix
@@ -47,6 +47,6 @@ buildPythonPackage (finalAttrs: {
     description = "Script to clone virtualenvs";
     mainProgram = "virtualenv-clone";
     license = lib.licenses.mit;
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ skohtv ];
   };
 })

--- a/pkgs/development/python-modules/virtualenv-clone/default.nix
+++ b/pkgs/development/python-modules/virtualenv-clone/default.nix
@@ -7,7 +7,7 @@
   setuptools,
 }:
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "virtualenv-clone";
   version = "0.5.7";
   pyproject = true;
@@ -15,7 +15,7 @@ buildPythonPackage rec {
   src = fetchFromGitHub {
     owner = "edwardgeorge";
     repo = "virtualenv-clone";
-    rev = version;
+    tag = finalAttrs.version;
     hash = "sha256-qrN74IwLRqiVPxU8gVhdiM34yBmiS/5ot07uroYPDVw=";
   };
 
@@ -23,11 +23,11 @@ buildPythonPackage rec {
 
   postPatch = ''
     substituteInPlace tests/__init__.py \
-      --replace-fail "'virtualenv'" "'${virtualenv}/bin/virtualenv'" \
+      --replace-fail "'virtualenv'" "'${lib.getExe virtualenv}'" \
       --replace-fail "'3.9', '3.10']" "'3.9', '3.10', '3.11', '3.12', '3.13', '3.14']" # if the Python version used isn't in this list, tests fail
 
     substituteInPlace tests/test_virtualenv_sys.py \
-      --replace-fail "'virtualenv'" "'${virtualenv}/bin/virtualenv'"
+      --replace-fail "'virtualenv'" "'${lib.getExe virtualenv}'"
 
     # PermissionError: [Errno 13] Permission denied: '/tmp/test_fixup_pth_file.pth'
     # Unable to reproduce.
@@ -49,4 +49,4 @@ buildPythonPackage rec {
     license = lib.licenses.mit;
     maintainers = [ ];
   };
-}
+})

--- a/pkgs/development/python-modules/virtualenv-clone/default.nix
+++ b/pkgs/development/python-modules/virtualenv-clone/default.nix
@@ -46,6 +46,7 @@ buildPythonPackage (finalAttrs: {
     homepage = "https://github.com/edwardgeorge/virtualenv-clone";
     description = "Script to clone virtualenvs";
     mainProgram = "virtualenv-clone";
+    changelog = "https://github.com/edwardgeorge/virtualenv-clone/blob/${finalAttrs.src.tag}/changelog.md";
     license = lib.licenses.mit;
     maintainers = with lib.maintainers; [ skohtv ];
   };

